demo/dynamic_image.cpp

```diff
@@ -0,0 +1,74 @@
+/*
+    Copyright 2019 WerWolv
+    Copyright 2019 p-sam
+    Copyright 2020-2021 natinusala
+
+    Licensed under the Apache License, Version 2.0 (the "License");
+    you may not use this file except in compliance with the License.
+    You may obtain a copy of the License at
+
+        http://www.apache.org/licenses/LICENSE-2.0
+
+    Unless required by applicable law or agreed to in writing, software
+    distributed under the License is distributed on an "AS IS" BASIS,
+    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+    See the License for the specific language governing permissions and
+    limitations under the License.
+*/
+
+#include "dynamic_image.hpp"
+
+DynamicImage::DynamicImage()
+{
+    // Load the XML file and inflate ourself with its content
+    this->inflateFromXMLRes("xml/views/dynamic_image.xml");
+
+    // Forward Image XML attributes
+    this->forwardXMLAttribute("scalingType", this->image);
+    this->forwardXMLAttribute("image", this->image);
+    this->forwardXMLAttribute("focusUp", this->image);
+    this->forwardXMLAttribute("focusRight", this->image);
+    this->forwardXMLAttribute("focusDown", this->image);
+    this->forwardXMLAttribute("focusLeft", this->image);
+    this->forwardXMLAttribute("imageWidth", this->image, "width");
+    this->forwardXMLAttribute("imageHeight", this->image, "height");
+
+    // Register a click action for this button
+    BRLS_REGISTER_CLICK_BY_ID("image", this->onImageClicked);
+}
+
+bool DynamicImage::onImageClicked(brls::View* view)
+{
+    // Retrieve the image size
+    int height = this->image->getHeight();
+    int width = this->image->getWidth();
+
+    // Generate the pixel data for a new image
+    size_t bufferSize = height * width * 4; // 4 bytes per pixel (RGBA)
+    unsigned char* imageData = (unsigned char*)calloc(1, bufferSize);
+
+    for (int y = 0; y < height; y++)
```

Comment:
Do you think you could randomize it?

And also make it so that it's "populated" when the view is created, so that it doesn't start all white?

Reply:
Trying to run the image generation in the RandomRGBAImage constructor causes a crash, I'm pretty sure it's because this->image->getHeight() / width() returns 0, hence it tries to allocate 0 bytes. At what stage is it safe to get the dimensions?

Reply:
You don't need the image width or height, you can just generate an image on a 256x256 grid and borealis will upscale it